### PR TITLE
Fix Shader default_texture_param not being fed to render code.

### PR DESCRIPTION
**Pull request by AlexSabourinDev**
Observed when assigning gradient to visual shader.

See rasterizer_storage_rd.h

default_texture_param map in both Shader and ShaderData. Rendering code reads from ShaderData map but not Shader map. Shader map however is not being fed to ShaderData map, causing rendering code to never retrieve the appropriate texture.
.
Fixed by removing Shader default_texture_param and only having the data owned by ShaderData. Following the premise of RasterizerStorageRD::shader_get_param_default.

Minimal Repro
[debug.zip](https://github.com/godotengine/godot/files/5079609/debug.zip)

### CODE REVIEW

**Review by clayjohn — APPROVED** [review]
Looks good to me. Seems like an oversight that ``default_texture_params`` was included both in the shader and in the ShaderData struct. 

Still needs @reduz' approval before merging.

**Comment by Chaosus** [comment]
Supersedes by #42046, @reduz said that is more clear implementation, thanks for the attempt anyway!

**Pull request closed without merging**
